overlord/snapstate/target.go

```diff
@@ -392,6 +392,12 @@ func installActionForStoreTarget(t StoreSnap, opts Options, enforcedSets func()
 		// caller provided some validation sets, nothing to do but send them
 		// to the store
 		action.ValidationSets = t.RevOpts.ValidationSets
+
+		// the channel here should be cleared out. if the validation sets that
+		// we are sending require a specific revision, we don't know if that
+		// revision will be part of any requested channel. the caller still
+		// might choose to track any channel in the RevisionOptions.
+		action.Channel = ""
```

Comment:
@alfonsosanchezbeato that is why we clear out the channel here and below when we have enforced validation sets

Reply:
Discussed offline: if a validation set requires a revision for a snap it does not matter if it was never published in the tracked channel of an already installed snap, it will be the one installed anyway.